src/main/java/com/softserve/paymentservice/service/PaymentService.java

```diff
@@ -0,0 +1,93 @@
+package com.softserve.paymentservice.service;
+
+import com.softserve.paymentservice.model.Card;
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.model.Charge;
+import com.stripe.model.Customer;
+import com.stripe.model.Refund;
+import com.stripe.model.Token;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class PaymentService {
+
+    @Value("${stripe.key.secret}")
+    private String stripeSecretKey;
+
+
+    public Boolean pay(int amount, Card card, String cvc) throws StripeException {
+        return getCharge(amount, card, cvc).getPaid();
+    }
+
+    public Boolean checkingPayment(Card card, String cvc) throws StripeException {
+        Charge charge = getCharge(100, card, cvc);
+        Refund refund = new Refund();
+        if (charge.getPaid()) {
+            boolean successfulRefund = false;
+            while (!successfulRefund) {
+                Map<String, Object> refundParameters = new HashMap<>();
+                refundParameters.put("charge", charge.getId());
+                refund = Refund.create(refundParameters);
+                if (refund.getStatus().equalsIgnoreCase("succeeded")) {
+                    successfulRefund = true;
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private Charge getCharge(int amount, Card card, String cvc) throws StripeException {
+
+        String customerId = getStripeCustomerId(card.getUserId());
+        String cardId = getSourceId(customerId, getCardMap(card, cvc));
+
+        Map<String, Object> chargeParams = new HashMap<>();
+        chargeParams.put("amount", amount);
+        chargeParams.put("currency", "EUR");
+        chargeParams.put("customer", customerId);
+        chargeParams.put("card", cardId);
+
+        Charge charge = Charge.create(chargeParams);
+        return charge;
+    }
+
+
+    private String getStripeCustomerId(UUID userId) throws StripeException {
+        Stripe.apiKey = stripeSecretKey;
```

Comment:
Can the apiKey be changed? As I see no, so it should be set once during application startup.